raiden-ts/src/raiden.ts

```diff
@@ -652,21 +662,25 @@ export class Raiden {
 
     const openTxHash = await openPromise;
     onChange?.({ type: EventTypes.OPENED, payload: { txHash: openTxHash } });
-
-    await this.state$
-      .pipe(
-        pluckDistinct('channels', channelKey(meta), 'state'),
-        first((state) => state === ChannelState.open),
-      )
-      .toPromise();
+    await openConfirmedPromise;
     onChange?.({ type: EventTypes.CONFIRMED, payload: { txHash: openTxHash } });
 
+    let depositTxHash: Hash | undefined;
     if (depositPromise) {
-      const depositTx = await depositPromise;
-      onChange?.({ type: EventTypes.DEPOSITED, payload: { txHash: depositTx } });
+      depositTxHash = await depositPromise;
+      onChange?.({ type: EventTypes.DEPOSITED, payload: { txHash: depositTxHash } });
       await postPromise;
     }
 
+    if (confirmConfirmation) {
+      // wait twice confirmationBlocks for deposit or open tx
+      await waitConfirmation(
+        await this.deps.provider.getTransactionReceipt(depositTxHash ?? openTxHash),
+        this.deps,
+        this.config.confirmationBlocks * 2,
```

Comment:
Just a nitpick, because that confused me, and I needed to inspect the
`waitConfirmation` function to understand it. Becaus we did already await the
`openPromise` which already includes 1x confirmation block numbers on top of the
tx-block. And now we wait again for 2x confirmation blocks. This works becuase
the function adds this number to the block number of the transaction receipt and
waits for this block. So for example if you call this function very late when
this block has been already mined it would resolve immediately. Maybe you could
put this detail into the function description? So a "hover" in the IDE already
reveals this secret.

Reply:
:heavy_check_mark: